### ISSUE TITLE
Memory access violation on Unity games (I am Bread, Stick Fight) (327890 674940) 

### ISSUE THREAD

**Issue by timschumi** [ValveSoftware/Proton#537]
I am experiencing crashes on Unity based games (only tested on "I am Bread" and "Stick Fight: The Game" so far), caused by an memory access violation right after starting the game.

I'll attach a few crash logs from both games (which are automatically generated by Unity, I can't attach the dumps because of a GitHub limitation) and my system/driver information (I added the information that is inside the square brackets).

[iambread_1_error.log](https://github.com/ValveSoftware/Proton/files/2320704/iambread_1_error.log)
[iambread_1_output_log.txt](https://github.com/ValveSoftware/Proton/files/2320705/iambread_1_output_log.txt)
[iambread_2_error.log](https://github.com/ValveSoftware/Proton/files/2320706/iambread_2_error.log)
[iambread_2_output_log.txt](https://github.com/ValveSoftware/Proton/files/2320707/iambread_2_output_log.txt)
[stickfight_1_error.log](https://github.com/ValveSoftware/Proton/files/2320708/stickfight_1_error.log)
[stickfight_1_output_log.txt](https://github.com/ValveSoftware/Proton/files/2320709/stickfight_1_output_log.txt)
[stickfight_2_error.log](https://github.com/ValveSoftware/Proton/files/2320710/stickfight_2_error.log)
[stickfight_2_output_log.txt](https://github.com/ValveSoftware/Proton/files/2320711/stickfight_2_output_log.txt)
[system_information.txt](https://github.com/ValveSoftware/Proton/files/2320712/system_information.txt)

**Comment by timschumi** [comment]
I've seen reports across the web that Stick Fight works on Manjaro KDE (which has a base similiar to Arch Linux) with Linux Kernel 4.14.65. I'll test a few configurations (different kernel versions and/or Proton versions) and see if I can achieve different results.

**Comment by timschumi** [comment]
Followup to my last comment: Neither downgrading to the mentioned kernel nor using another Proton version (3.7-4 in this case) helped solving the problem. However, I was able to confirm that both games are starting successfully using wine-3.14 from the Arch Linux repositories.

**Comment by t1m0thyj** [comment]
I encountered the same error trying to run Stick Fight using Steam Play in Linux Mint 19, on a laptop with Intel HD 520 integrated graphics.

**Comment by pmykc1** [comment]
I also have this very same problem.

Here is my error log for the unity game Overcooked:
https://gist.github.com/pmykc1/99f0a67b61fdb6cd4765f030147bf372
my system details are here:
https://gist.github.com/pmykc1/5c6eedd49b631980f3718aa3dc530b98

Having looked around at a few system specs, it seems that ALL of the people who can run unity games have Nvidia graphics cards, and all of us who cannot have Intel graphics cards. I think this is where the problem lies! But I am not knowledgeable enough to sleuth more than that.

**Comment by timschumi** [comment]
@pmykc1 I have a friend who I think has a relatively old Nvidia card and he said that Stick Fight is crashing for him as well. I'll try and get some logs and system information from him, that may help.

**Comment by pmykc1** [comment]
The upgraded graphics drivers not included in 18.04 have allowed me to run unity games:
`sudo add-apt-repository ppa:paulo-miguel-dias/mesa`
`sudo apt dist-upgrade`
`sudo apt install mesa-vulkan-drivers mesa-vulkan-drivers:i386`

**Comment by timschumi** [comment]
@pmykc1 Huh, doesn't work for me (at least on Stick Fight)

**Comment by bla-rs** [comment]
+1 for access violation with Unity.

Over here i can start the game "Battlerite Royale" just fine (Ubuntu 18.04, latest Nvidia drivers) and the performance/appearance looks very good but as soon as i hit something networking related, the game crashes. See the logs below for details.

[error.log](https://gist.github.com/bla-rs/f41e71fabd949fdb8d7989b1d4d33a02)
[output_log.txt](https://gist.github.com/bla-rs/378c1b9624df13940e73bd314a70ecd7)

**Comment by PCFreak2000** [comment]
Getting the same issues here on Stick Fight, with an Nvidia GPU though.

[error.log](https://github.com/ValveSoftware/Proton/files/2453989/error.log)
[output_log.txt](https://github.com/ValveSoftware/Proton/files/2453990/output_log.txt)
[system_information.txt](https://github.com/ValveSoftware/Proton/files/2453993/system_information.txt)

**Comment by kisak-valve** [comment]
Hello @PCFreak2000, in your case your hardware is not capable of directx 11. You could try some of the [Proton runtime options](https://github.com/ValveSoftware/Proton#runtime-config-options) or [unity 5.6 options](https://docs.unity3d.com/560/Documentation/Manual/CommandLineArguments.html).

From that, maybe adding `-force-d3d9` or `-force-glcore33` in the game's launch options would help.

**Comment by timschumi** [comment]
Forcing D3D resolved the problems for me. I assume that Unity does query our hardware and assumes that we support some Graphics API (or some extension of it). The setting for the used Graphics API seems to persist when restarting the game, even if the launch option is then removed, so starting the game with the launch option once is enough.

**Comment by feimosi** [comment]
`PROTON_USE_WINED3D11=1 %command%` in Launch Options did the trick for me.

**Comment by gwemmie** [comment]
Nope, not only Intel. I'm AMD and I too had this issue, and @feimosi's workaround fixed it for me, too. However, now I'm getting extremely dark 3D objects to the point that I can only even see the very brightest bits of them, making the whole thing almost just a black screen with music.

**Comment by douglasg14b** [comment]
Also occurs with Dig or Die.

`PROTON_USE_WINED3D11=1 %command%` did resolve it. Though closing dig or die causes the steam client to crash...

**Comment by timschumi** [comment]
I can't replicate this anymore, neither using the most recent Proton 3.7 build, nor on any other version (using the default settings and without any launch arguments).

Is anybody else still having this issue? Otherwise, I'm going to assume that this has been fixed somewhere along the way.

**Comment by Turantel** [comment]
I was getting this issue with The Forest. feimosi's solution helped out with getting it running again
